server/src/immich/app.guard.ts

```diff
@@ -93,12 +93,12 @@ export class AppGuard implements CanActivate {
     const req = context.switchToHttp().getRequest<AuthRequest>();
 
     const authDto = await this.authService.validate(req.headers, req.query as Record<string, string>);
-    if (authDto.isPublicUser && !isSharedRoute) {
+    if (authDto.sharedLink && !isSharedRoute) {
```

Comment:
What is the alternative mechanism to determine if a user is a public user?

Reply:
auth via a shared link implies the user is a "public user".